Tools/TestWebKitAPI/Tests/WebKitCocoa/LocalStoragePersistence.mm

```diff
@@ -105,13 +106,12 @@ - (void)webView:(WKWebView *)webView didFinishNavigation:(WKNavigation *)navigat
     NSURLRequest *request = [NSURLRequest requestWithURL:[[NSBundle mainBundle] URLForResource:@"local-storage-process-crashes" withExtension:@"html" subdirectory:@"TestWebKitAPI.resources"]];
     [webView loadRequest:request];
 
-    receivedScriptMessage = false;
-    TestWebKitAPI::Util::run(&receivedScriptMessage);
-    EXPECT_WK_STREQ(@"local:storage", [lastScriptMessage body]);
+    TestWebKitAPI::Util::waitForConditionWithLogging([&] {
+        return [[lastScriptMessage body] isEqualToString:@"session:storage"];
+    }, 10, @"Warning: expected 'session:storage' message after page load");
 
-    receivedScriptMessage = false;
-    TestWebKitAPI::Util::run(&receivedScriptMessage);
-    EXPECT_WK_STREQ(@"session:storage", [lastScriptMessage body]);
+    EXPECT_EQ([receivedMessages count], 2U);
+    EXPECT_WK_STREQ(@"local:storage", [receivedMessages firstObject]);
```

Comment:
Does the content of the other item in this array not matter?

Reply:
Ah, so we do care about the first value, but the `waitForConditionWithLogging` above is sufficient to verify that the second value is `session:storage`